lib/Infer/Pruning.cpp

```diff
@@ -119,6 +119,32 @@ std::pair<llvm::APInt, llvm::APInt> knownBitsNarrowing
   return {KnownNotZero, KnownNotOne};
 }
 
+std::unordered_map<Inst *, llvm::APInt>
+PruningManager::computeInputRestrictions() {
+  std::unordered_map<Inst *, std::pair<llvm::APInt, llvm::APInt>> Seen;
```

Comment:
you don't need to change it here but overall you should avoid using a std::pair for same-typed things like this, and instead pick an implementation which assigns appropriate names to things. here two separate variables would work, or else a struct with members called Zero and One or similar